packages/shared-components/src/components/phone-number/PhoneNumber.tsx

```diff
@@ -0,0 +1,140 @@
+import { BodyShort, Select, TextField } from '@navikt/ds-react';
+import {
+  CustomLabels,
+  FieldSize,
+  formatPhoneNumber,
+  removeAllSpaces,
+  TEXTS,
+} from '@navikt/skjemadigitalisering-shared-domain';
+import clsx from 'clsx';
+import { useCallback, useEffect, useState } from 'react';
+import { getAreaCodes } from '../../api/common-codes/area-codes';
+import { useComponentUtils } from '../../context/component/componentUtilsContext';
+import { PhoneNumberObject } from '../../formio/components/core/phone-number/PhoneNumber';
+import { FieldsetErrorMessage } from '../error/FieldsetErrorMessage';
+import { usePhoneNumberStyles } from './styles';
+import { getInitialPhoneNumber } from './utils';
+
+interface Props {
+  label: React.ReactNode;
+  description?: React.ReactNode;
+  className?: string;
+  value: string | PhoneNumberObject;
+  onChange: (value: any) => void;
+  error?: React.ReactNode;
+  readOnly?: boolean;
+  fieldSize?: FieldSize;
+  required?: boolean;
+  customLabels?: CustomLabels;
+  showAreaCode: boolean;
+}
+
+type AreaCode = { value: string; label: string };
+
+type PhoneNumber = {
+  areaCode: string;
+  number: string;
+};
+
+const PhoneNumber = ({ value, onChange, showAreaCode, error }: Props) => {
+  const { appConfig } = useComponentUtils();
+  const [areaCodes, setAreaCodes] = useState<AreaCode[]>([]);
+  const [fetchError, setFetchError] = useState<string | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [phoneNumber, setPhoneNumber] = useState<PhoneNumber | string>(getInitialPhoneNumber(value, showAreaCode));
+  const styles = usePhoneNumberStyles();
+  const defaultAreaCode = '+47';
+
+  const fetchAreaCodes = useCallback(async () => {
+    setFetchError(null);
+    setLoading(true);
+    try {
+      const data = await getAreaCodes(appConfig.fyllutBaseURL);
+      setAreaCodes(data);
+    } catch (error) {
+      setFetchError(error instanceof Error ? error.message : String(error));
+      setAreaCodes([]);
+    } finally {
+      setLoading(false);
+    }
+  }, [appConfig.fyllutBaseURL]);
+
+  useEffect(() => {
+    if (showAreaCode) {
+      fetchAreaCodes();
+    }
+  }, [fetchAreaCodes, showAreaCode, appConfig.fyllutBaseURL]);
+
+  if (showAreaCode && (loading || !areaCodes.length)) return null;
+
+  function handleChange(value: string, key: string) {
+    if (showAreaCode) {
+      const updated: PhoneNumber = {
+        ...(typeof phoneNumber === 'object' && phoneNumber ? phoneNumber : { areaCode: '+47', number: '' }),
+        [key]: removeAllSpaces(value),
```

Comment:
Vi ønsker å gjøre dette for `onChange`, men vi vil helst ikke endre på input verdien som brukeren sitter og fyller ut før eventuelt onBlur, og aller helst ikke før du har forlatt siden.